src/Approximations/decompositions.jl

```diff
@@ -226,7 +226,13 @@ function decompose(S::LazySet{N};
     n = dim(S)
     result = Vector{LazySet{N}}()
 
-    if directions != nothing
+    if set_type == LinearMap
+        block_start = 1
+        @inbounds for bi in blocks
+            push!(result, project(S, block_start:(block_start + bi - 1), n))
+            block_start += bi
+    end
+    elseif directions != nothing
```

Comment:
I would not add another condition here.
Just use the existing code and pass `set_type` to the `project` method.